src/utilities.jl

```diff
@@ -88,8 +88,26 @@ function checkindprsk(k::Integer)
 end
 
 """
-    replicate(f::Function, n::Integer)
+    replicate(f::Function, n::Integer, use_threads=false)
 
 Return a vector of the values of `n` calls to `f()` - used in simulations where the value of `f` is stochastic.
+
+Note that if `f()` is not thread-safe or depends on a non thread-safe RNG,
+    then you must set `use_threads=false`. Also note that ordering of replications
+    is not guaranteed when `use_threads=true`, although the replications are not
+    otherwise affected for thread-safe `f()`.
 """
-replicate(f::Function, n) = [f() for _ in Base.OneTo(n)]
+function replicate(f::Function, n::Integer; use_threads=false)
+    if use_threads
+        # get the type
+        rr = f()
+        # pre-allocate
+        results = [rr for _ in Base.OneTo(n)]
+        Threads.@threads for idx = 2:n
+            results[idx] = f()
+        end
+    else
+        results = [f() for _ in Base.OneTo(n)]
```

Comment:
On this line I meant to add `@showprogress` before the `[`.  That is why the `ProgressMeter` package is included with this package.  Would you be willing to add it to this PR?

Reply:
Yes -- I'll have to see how interacts with threading, but at least for single-threaded case, it's easy.

Reply:
I followed the ProgressMeter example for a threaded loop exactly, but the bar still redraws itself weirdly from time to time. I'm wondering if it's interacting with some logging statement somewhere, but that should impact the non threaded version equally.